igor-web/src/main/groovy/com/netflix/spinnaker/igor/docker/DockerRegistryCacheV2KeysMigration.java

```diff
@@ -0,0 +1,180 @@
+        return running.get();
+    }
+
+    @PostConstruct
+    void run() {
+        running.set(true);
+        try {
+            scheduler.createWorker().schedule(this::migrate);
+        } finally {
+            running.set(false);
+        }
+    }
+
+    private void migrate() {
+        log.info("Starting migration");
+
+        List<String> oldKeys = redis.withMultiClient(this::getV1Keys);
+        log.info("Migrating {} v1 keys", oldKeys.size());
+
+        int batchSize = properties.getRedis().getDockerV1KeyMigration().getBatchSize();
+        for (List<String> oldKeyBatch : Iterables.partition(oldKeys, batchSize)) {
+            // For each key: Check if old exists, if so, copy to new key, set ttl on old key, remove ttl on new key
+            migrateBatch(oldKeyBatch);
+        }
+    }
+
+    private void migrateBatch(List<String> oldKeys) {
+        int expireSeconds = (int) Duration.ofDays(properties.getRedis().getDockerV1KeyMigration().getTtlDays()).getSeconds();
+        redis.withCommandsClient(c -> {
+            for (String oldKey : oldKeys) {
+                String newKey = convertToNewFormat(oldKey);
+                if (c.exists(newKey)) {
+                    // Nothing to do here, just move on with life
+                    continue;
+                }
+
+                Map<String, String> value = c.hgetAll(oldKey);
+                c.hmset(newKey, value);
+                c.expire(oldKey, expireSeconds);
+            }
+        });
+    }
+
+    private List<String> getV1Keys(MultiKeyCommands client) {
+        // TODO rz - dyno does not yet support `scan` interface; exposed as `dyno_scan`
+        if (redis instanceof DynomiteClientDelegate) {
+            return keys((DynoJedisClient) client);
+        }
+        return keys((Jedis) client);
+    }
+
+    private List<String> keys(DynoJedisClient dyno) {
+        List<String> keys = new ArrayList<>();
+
+        String pattern = oldIndexPattern();
+        CursorBasedResult<String> result;
+        do {
+            result = dyno.dyno_scan(pattern);
+            keys.addAll(result.getResult().stream().filter(this::isOldKey).collect(Collectors.toList()));
+        } while (!result.isComplete());
+
+        return keys;
+    }
+
+    private List<String> keys(Jedis jedis) {
+        List<String> keys = new ArrayList<>();
+
+        ScanParams params = new ScanParams().match(oldIndexPattern()).count(1000);
+        String cursor = ScanParams.SCAN_POINTER_START;
+
+        ScanResult<String> result;
+        do {
+            result = jedis.scan(cursor, params);
+            keys.addAll(result.getResult().stream().filter(this::isOldKey).collect(Collectors.toList()));
+        } while (!result.getStringCursor().equals("0"));
+
+        return keys;
+    }
+
+    private boolean isOldKey(String key) {
+        // Target v1 keys. They include repository URLs, which can include ports, so eq/gt 6 parts.
+        return key.split(":").length >= 6;
```

Comment:
are we certain this can't fire on new keys? For instance, we could check that the 2nd token is not "v2" as a safety